### PR TITLE
Ensure outgoing edge map is sorted by upstream label

### LINKED ISSUE #64: Outgoing edges of nodes should be sorted based on when a call `.stream()` created them
```python
i = ffmpeg.input("f1.mp4")
ref = ffmpeg.input("f2.mp4")
# BTW `filter_multi_output` is not in __all__, needs #65
s2ref = ffmpeg.filter_([i, ref], "scale2ref").node
scaled = s2ref[0]
ref2 = s2ref[1]
```
https://ffmpeg.org/ffmpeg-all.html#scale2ref

[](https://postimg.org/image/f9k9b7kmd/)

You would expect the output of `scale2ref` labeled 0 (`scaled`) to be the first output of `scale2ref` (because `.stream()` was called first), and 1 (`ref2`) to be the second. However, this doesn't always happen: it's totally up to `topo_sort()`.

I'll try to fix it and send a PR.

### PR DESCRIPTION
**Pull request by Depau**
Closes #64

### CODE REVIEW

File: ffmpeg/_utils.py [hunk 1]
```diff
@@ -41,3 +42,8 @@ def escape_chars(text, chars):
     for ch in chars:
         text = text.replace(ch, '\\' + ch)
     return text
+
+
+def sort_ordereddict(d):
+    for key in sorted(d.keys()):
+        d[key] = d.pop(key)
```
Comment (Depaulicious) [inline ffmpeg/_utils.py]: This moves the key to the end of the dict:
- https://docs.python.org/3.6/library/collections.html#collections.OrderedDict

**Comment by kkroening** [comment]
Is there a test case that demonstrates what this fixes? I haven't been able to reproduce the bug.

**Comment by Depau** [comment]
No, there isn't. I'll pin this tab so I don't forget about it.

**Comment by kkroening** [comment]
I'm still hesitant to merge this if we don't have an actual example of a case that it affects. I'll merge it once we have an example.

**Comment by kkroening** [comment]
If we can't reproduce the issue I'm going to close this.

If the issue happens again in a reproducible manner, we can resurrect the branch.

**Comment by komar007** [comment]
Hi,

I have just reproduced this issue in a case where 2 outputs from scale2ref are in wrong order (always the same) no matter what I do. I have also verified that this pull request fixes the issue.
Unfortunately the reproduction program is huge with over 230 edges and it has plenty of dependencies, so I cannot post it here.

Why was it not merged?
I don't have the full understanding of the code, but it looks like node's outgoing edges are not sorted (and represented as a dict). This may work for the split filter, but certainly not for something like scale2ref, where the order of outputs is crucial.
If, on the contrary, the intention was for the outgoing edges to be sorted by label, then relying on a data structure (dict) which is by definition iterated in arbitrary order is an error.
This pull request clearly fixes a data structure design issue in the code. It shouldn't require reproduction to justify merging it.

Thanks for the great work! ffmpeg-python has been very useful to me so far. It would be great to see this bug fixed in the upstream.

**Pull request closed without merging**